src/main/java/org/gridsuite/study/server/service/NetworkModificationTreeService.java

```diff
@@ -712,12 +713,33 @@ private void invalidateNodeProper(NodeEntity child, InvalidateNodeInfos invalida
     public void updateBuildStatus(UUID nodeUuid, BuildStatus buildStatus) {
         List<UUID> changedNodes = new ArrayList<>();
         UUID studyId = getStudyUuidForNodeId(nodeUuid);
+        NodeEntity nodeEntity = getNodeEntity(nodeUuid);
 
-        nodesRepository.findById(nodeUuid).ifPresent(n -> repositories.get(n.getType()).updateBuildStatus(nodeUuid, buildStatus, changedNodes));
+        BuildStatus newNodeStatus;
+        if (buildStatus.isBuilt()) {
+            NodeEntity previousBuiltNode = doGetLastParentNodeBuilt(nodeEntity);
+            BuildStatus previousBuiltNodeStatus = repositories.get(previousBuiltNode.getType()).getBuildStatus(previousBuiltNode.getIdNode());
```

Comment:
Add a method with an explicit name to calculate the max

Reply:
Done